sklearn/utils/validation.py

```diff
@@ -535,7 +535,7 @@ def _ensure_sparse_format(
 
     if accept_sparse is False:
         raise TypeError(
-            "A sparse matrix was passed, but dense data is required. Use X.toarray() "
+            "A sparse matrix was passed, but dense data is required. Use '.toarray()' "
```

Comment:
And I assume that it can be a sparse matrix or array.

Reply:
Sorry I don't follow?

Reply:
I suppose this is about sparse array vs sparse matrix. More context in https://github.com/scikit-learn/scikit-learn/issues/26418.

Maybe use something like "sparse data" or `sparse_container.__class__` in the error message?

Reply:
Ah thank you, I did not know the difference.

Reply:
I've used 'sparse data' as I think that is enough info but happy to change.

Reply:
I've added `input_name`, if present, to the error message as sometimes more than one argument will be checked (e.g., X and y) and this can help. Happy to change back though.

cc @glemaitre in case this needs re-review.